src/cargo/commands.rs

```diff
@@ -0,0 +1,168 @@
+
+use cargo::command_bridge::CommandBridge;
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+
+use error::*;
+
+pub fn retrieve_metadata(manifest_path: &Path) -> CommandBridge {
+    let mut command = CommandBridge::new("cargo");
+    command.arg("metadata")
+        .arg("--manifest-path")
+        .arg(manifest_path)
+        .arg("--no-deps")
+        .arg("--format-version")
+        .arg("1");
+    command
+}
+
+pub fn generate_analysis(manifest_path: &PathBuf, is_verbose: bool) -> Result<CommandBridge> {
+
+    check_manifest_path_points_to_cargo_toml(manifest_path)?;
+
+    const RLS_TARGET : &str = "target/rls";
+    let target_dir = manifest_path
+        .parent()
+        .expect("Unreachable. If not, fill a bug about `check_manifest_path_points_to_cargo_toml`")
+        .join(RLS_TARGET);
+
+    let mut command = CommandBridge::new("cargo");
+
+    command
+        .arg("check")
+        .arg("--manifest-path")
+        .arg(&manifest_path)
+        //.env("CARGO_TARGET_DIR", target_dir) // FIXME compiles
+        .env("RUSTFLAGS", "-Z save-analysis")
+        .stderr(Stdio::piped())
+        .stdout(Stdio::null());
+
+    if is_verbose {
+        command.arg("--verbose");
```

Comment:
Hmmm this would cause command to be consumed if you did what I mentioned. You'd have to do a let command = command.arg("--verbose");` here

Reply:
👍